### PR TITLE
SEO delete category in URL product scheme during installation

### PR DESCRIPTION
**Pull request by jeckyl**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 9.0.x
| Description?      | Replace {category:/}{id}{-:id_product_attribute}-{rewrite}{-:ean13}.html by {id}{-:id_product_attribute}-{rewrite}{-:ean13}.html in product url sheme. Removal of category reference in product URLs during PrestaShop installation. This has no effect on PrestaShop updates. It simplifies product URLs, allowing you to simply add the category if you wish without any further action, and avoids the need to redirect old URLs including categories to 301 if the user wishes to remove this data.
| Type?             | improvement
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| UI Tests | https://github.com/florine2623/testing_pr/actions/runs/12070989574 ✅ 
| How to test?      | This change modifies the structure of product URLs when PrestaShop is installed, and only when it's installed. It doesn't change anything when you upgrade, or at least it shouldn't modify the information when you upgrade.
| Fixed issue or discussion?     | Fixes https://github.com/PrestaShop/PrestaShop/discussions/37437
| Sponsor company   | Mediacom87.fr

### CODE REVIEW

**Comment by ps-jarvis** [comment]
Hi, thanks for this contribution!

I found some issues with the Pull Request description:

* The `category` should be one of these: `FO`, `BO`, `CO`, `IN`, `WS`, `TE`, `LO`, `ME` or `PM`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))

Would you mind having a look at it? This will help us understand how interesting your contribution is, thank you very much!

(Note: this is an automated message, but answering it will reach a real human)

**Review by jf-viguier — APPROVED** [review]

**Comment by kpodemski** [comment]
ping @PrestaShop/product-council @ibahloul-ps 

Thank you @jeckyl 

@ibahloul-ps please take a look at this PR and the Discussion related, this PR aims to change URLs from:

domain.com/category/1-product.html

to

domain.com/1-product.html

it follows the latest - I would say - trends, it doesn't impact existing installations and it would still be possible to configure the URLs to have categories if one wants to

**Review by Hlavtox — CHANGES_REQUESTED** [review]
I would personally go for `domain.com/product/1-product.html` or `domain.com/p/1-product.html`

That will allow us to also get rid of the `.html` suffix that's there.

**Comment by jeckyl** [comment]
> I would personally go for `domain.com/product/1-product.html` or `domain.com/p/1-product.html`
> 
> That will allow us to also get rid of the `.html` suffix that's there.

Except that doing so would break PrestaShop's native functionality, since you'd have to modify htaccess to manage the old form of product URLs.
What I'm proposing, on the other hand, is a simple way of implementing your URL structure if you wish.

**Comment by kpodemski** [comment]
> I would personally go for `domain.com/product/1-product.html` or `domain.com/p/1-product.html`
> 
> 
> 
> That will allow us to also get rid of the `.html` suffix that's there.

That's not what has been agreed

**Comment by Hlavtox** [comment]
@kpodemski If we are changing default product route, let's think for one more minute, that's all. I am not dwelling on anything, just a suggestion.

@jeckyl Why do you think it will need htaccess change?

**Comment by jeckyl** [comment]
> @jeckyl Why do you think it will need htaccess change?

If you base the creation of a store on a sub-folder in the URL, whatever it may be, if you want to remove it later, the old URLs will be in 404 error mode, just like when the URLs included the category.
It's better to leave it up to users to add whatever they want.

**Review by Touxten — APPROVED** [review]
I also likes https://github.com/Hlavtox  idea, surely in a future Major or we could also remove the ean13 ( NPM ) in new version

**Comment by LouisAUTHIE** [comment]
I would be in favor of a ean13 removal that leads to useless and very long urls

**Review by LouisAUTHIE — CHANGES_REQUESTED** [review]
Maybe go further and remove the ean13. But the rest is clear and great thanks @jeckyl

**Comment by jeckyl** [comment]
> Maybe go further and remove the ean13. But the rest is clear and great thanks @jeckyl

The removal or presence of the EAN13 in the URL only affects a redirection automatically handled by PrestaShop and doesn't require any technical expertise to intervene on this change, so I haven't bothered to broaden the discussion on this subject.
The discussion is interesting and can be opened on the subject to gather as many opinions as possible and make the best decision as to whether to remove it or keep it.

**Review by Touxten — APPROVED** [review]

**Review by LouisAUTHIE — DISMISSED** [review]
Great job @jeckyl

**Comment by Hlavtox** [comment]
Guys, suggestion. I just tested dropping the .html at the end and it works perfectly.

See - [https://dev.trendo.cz/product/2-osklivy-produkt](https://dev.trendo.cz/product/2-osklivy-produkt)

WDYT about this last change?

**Comment by jeckyl** [comment]
> Guys, suggestion. I just tested dropping the .html at the end and it works perfectly.
> 
> See - https://dev.trendo.cz/product/2-osklivy-produkt
> 
> WDYT about this last change?

Please open a discussion on this point to really explain the interest and benefits of changing URLs for this form.

**Comment by Touxten** [comment]
Hello Daniel!

I agree with you it would be necessary to check the bugs if there are with the category and cms urls.

This Pr is part of the Friends of Presta academy to have new contributors.

What do you think about validating it and we open a new discussion and @jeckyl can make a new PR

Regards

**Comment by jeckyl** [comment]
The real question is what's the difference and what's the point of replacing .html with /product/, an English term that can't be translated, and which will therefore potentially penalize all stores in other languages if the URLs had any importance at all.
Hence the interest in seriously starting a new discussion on this subject, as another person has opened a discussion about moving the ID into the URL.
If my PR is mainly technical and not SEO, any other changes are just cosmetics that can and should be discussed.

**Review by Hlavtox — DISMISSED** [review]

**Comment by Hlavtox** [comment]
I don't want to block it, I will open a new discussion about my proposed change. :-)

I want to remove the .html for the sake of consistency. It's just a leftover from old times.

And by the way, it won't be that difficult to make routes translatable. ;-)

**Review by florine2623 — COMMENTED** [review]
Hello @Progi1984 ,

The auto tests needs to be updated right ?

https://github.com/florine2623/testing_pr/actions/runs/12070989574

It seems like there's an issue in the Home Page.

Waiting for your feedback, thanks!

**Comment by Progi1984** [comment]
Discussed with @Touxten for fixing UI tests :)

**Review by Progi1984 — APPROVED** [review]
OK for UI Tests

**Comment by Progi1984** [comment]
@florine2623 COuld you restart only failed tests, please ?

**Review by florine2623 — APPROVED** [review]
Hello @Touxten ,

The URL is well updated. The category is not displayed anymore.

I can still add it if I want to in SEO & URL in the BO.

LGTM ✅ 
Thanks!

**PS : Before merging, it needs some more approvals.**

**Review by Hlavtox — APPROVED** [review]

**Review by kpodemski — CHANGES_REQUESTED** [review]
let's wait for @ibahloul-ps for the final approval

**Review by LouisAUTHIE — APPROVED** [review]

**Review by ibahloul-ps — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
thank you @jeckyl

**Comment by Hlavtox** [comment]
This PR causes a pretty serious SEO issue. Fixed by https://github.com/PrestaShop/autoupgrade/pull/1535.

**Pull request merged**
